cadc-download-manager-server/src/main/java/ca/nrc/cadc/dlm/server/ServerUtil.java

```diff
@@ -153,28 +155,177 @@ public static String getCodebase(HttpServletRequest request) {
      * @param request
      * @return
      */
-    public static List<String> getURIs(HttpServletRequest request) {
+    public static List<URI> getURIs(HttpServletRequest request) throws URISyntaxException {
+
+        List<URI> ret = new ArrayList<>();
+
         // internal repost
         String uris = request.getParameter("uris");
-
+        String[] uriParams;
         if (uris != null) {
-            return DownloadUtil.decodeListURI(uris);
+            uriParams = uris.split(" ");
+        } else {
+            // original post
+            uriParams = request.getParameterValues(PARAM_URI);
         }
 
-        // original post
-        String[] uriParams = request.getParameterValues(PARAM_URI);
-
-        List<String> ret = new ArrayList<>();
-
+        // process into a List<URI>
         if (uriParams != null) {
             for (String u : uriParams) {
                 if (StringUtil.hasText(u)) {
-                    ret.add(u);
+                    ret.add(new URI(u));
+                }
+            }
+        }
+
+        // In case nothing is passed in as 'uri' or 'uris,' check for
+        // deprecated parameters
+        if (ret.isEmpty()) {
+            handleDeprecatedParams(request);
+            // Could be populated if parameters are passed in at all.
+            ret = (List<URI>)request.getAttribute("uriList");
+        }
+
+        return ret;
+    }
+
+    /**
+     * Check for deprecated parameters in the request. Convert to the most current for Download Manager's API.
+     * Kept for backward compatibility.
+     * (last rev: July 21, 2020, s2739, HJ)
+     * @param request
+     */
+    public static void handleDeprecatedParams(HttpServletRequest request) {
+        // Check to see if URIs have already been parsed from the request parameters.
+        // If so, this is an internal dispatch/forward being processed
+        List<URI> uriList = (List<URI>)request.getAttribute("uriList");
```

Comment:
handleDeprecatedParams should only look in the request for param=value pairs (should not be looking at request attributes), turn them into URIs, and return them. get rid of any side effects like setting request attributes.

Reply:
done